org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WColorEditor.java

```diff
@@ -0,0 +1,268 @@
+	}
+
+	protected void addColorEditorMenu(WEditorPopupMenu popupMenu) {
+		Menuitem editor = new Menuitem();
+		editor.setAttribute("EVENT", WEditorPopupMenu.RESET_EVENT);
+		editor.setLabel(Util.cleanAmp(Msg.getMsg(Env.getCtx(), "Erase")).intern());
+		if (ThemeManager.isUseFontIconForImage())
+			editor.setIconSclass("z-icon-eraser");
+		else
+			editor.setImage(ThemeManager.getThemeResource("images/Erase16.png"));
+		editor.addEventListener(Events.ON_CLICK, popupMenu);
+		popupMenu.appendChild(editor);
+
+		editor = new Menuitem();
+		editor.setAttribute("EVENT", WEditorPopupMenu.COLOR_PICKER_EVENT);
+		editor.setLabel(Msg.getMsg(Env.getCtx(), "ColorPicker"));
+		if (ThemeManager.isUseFontIconForImage())
+			editor.setIconSclass("z-icon-pencil");
+		else
+			editor.setImage(ThemeManager.getThemeResource("images/ColorPicker16.png"));
+		editor.addEventListener(Events.ON_CLICK, popupMenu);
+		popupMenu.appendChild(editor);
+	}
+
+	public void onMenu(ContextMenuEvent evt)
+	{
+		if (WEditorPopupMenu.RESET_EVENT.equals(evt.getContextEvent()))
+		{
+			processNewValue(null);
+			colorbox.setValue(oldValue);
+
+		}
+		else if (WEditorPopupMenu.COLOR_PICKER_EVENT.equals(evt.getContextEvent()))
+		{
+			openColorPicker();
+		}
+	}
+
+	@Override
+	public EditorBox getComponent()
+	{
+		return (EditorBox) component;
+	}
+
+	@Override
+	public void setValue(Object value)
+	{
+		if (value == null)
+		{
+			oldValue = null;
+			getComponent().setText("");
+		}
+		else
+		{
+			oldValue = String.valueOf(value);
+			getComponent().setText(oldValue);
+		}
+		colorbox.setValue(oldValue);
+
+		fillTextbox();
+	}
+
+	private void fillTextbox() {
+		String style="background-color: transparent !important;";
+		if (!Util.isEmpty(oldValue, true))
+			style = "background: linear-gradient(to right, rgba(255,0,0,0) 50%, "
+					+ oldValue + " 50%) !important;";
+		String script = "jq('#"+getComponent().getTextbox().getUuid()+"').attr('style','"+style+"');";
+		if (Executions.getCurrent() != null)
+			Clients.response(new AuScript(script));
+		else if (getComponent().getDesktop() != null)
+			Executions.schedule(getComponent().getDesktop(), e -> Clients.response(new AuScript(script)), new Event("onFillTextBox"));
+	}
+
+	@Override
+	public Object getValue()
+	{
+		return getComponent().getText();
+	}
+
+	@Override
+	public String getDisplay()
+	{
+		return getComponent().getText();
+	}
+
+	@Override
+	public boolean isReadWrite() {
+		return getComponent().getButton().isEnabled();
+	}
+
+	@Override
+	public void setReadWrite(boolean readWrite) {
+		getComponent().getButton().setEnabled(readWrite);
+	}
+
+	public void onEvent(Event event)
+	{
+		if (Events.ON_CLICK.equalsIgnoreCase(event.getName()))
+		{
+			openColorPicker();
+		}
+		else if (Events.ON_CHANGE.equals(event.getName()) || Events.ON_OK.equals(event.getName()))
+		{
+			String newValue = getComponent().getTextbox().getValue();
+			processNewValue(newValue);
+		}
+		else
+		{
+			return;
+		}
+	}
```

Comment:
I think both event is no longer need here.

Reply:
done